x/txfees/keeper/fees.go

```
@@ -152,19 +150,27 @@ func (k Keeper) swapFeeToBaseDenom(
 	}
 
 	// Swap the coin to base denom
-	var (
-		tokenOutAmount = sdk.ZeroInt() // Token amount in base denom
-		route          = []poolmanagertypes.SwapAmountInRoute{{
-			PoolId:        feetoken.PoolID,
-			TokenOutDenom: baseDenom,
-		}}
-	)
-	err = osmoutils.ApplyFuncIfNoError(ctx, func(ctx sdk.Context) error {
-		tokenOutAmount, err = k.poolManager.RouteExactAmountIn(ctx, moduleAddr, route, takerFeeCoin, sdk.ZeroInt())
-		return err
-	})
+	route := []poolmanagertypes.SwapAmountInRoute{{
+		PoolId:        feetoken.PoolID,
+		TokenOutDenom: baseDenom,
+	}}
+
+	cacheCtx, write := osmoutils.NoEventCacheContext(ctx)
+	tokenOutAmount, err := k.poolManager.RouteExactAmountIn(cacheCtx, moduleAddr, route, takerFeeCoin, sdk.ZeroInt())
 	if err != nil {
-		return nil, sdk.Coins{takerFeeCoin}, nil
+		return nil, nil, fmt.Errorf("swap fee token: %w", err)
+	}
+	write()
+
+	// if we manage, we modify the last event to include the taker fee attribute
+	emittedEvents := cacheCtx.EventManager().Events()
+	if len(emittedEvents) > 0 && emittedEvents[len(emittedEvents)-1].Type == gammtypes.TypeEvtTokenSwapped {
```

seems a bit brittle if you depend on it being the last one
if something changes it will silently fail

---

not supposed to happen,
added logging just in case